net/Socket.hpp

```diff
@@ -1056,6 +1056,19 @@ class StreamSocket : public Socket,
         _readType(readType),
         _inputProcessingEnabled(true)
     {
+#if TEST_SERVERSOCKET_FAULT_INJECTION
```

Comment:
We should have a unit test that tests this using a normal unit test hook - two lines at the site, and then more work in test/

Reply:
as discussed above, OK

Reply:
revised.